love/src/Utils.js

```diff
@@ -1840,3 +1840,196 @@ export function trimString(string, length = 100) {
   }
   return string;
 }
+
+/**
+ * Function to parse HTML, generated by react-quill, to Jira Markdown.
+ * Check https://jira.atlassian.com/secure/WikiRendererHelpAction.jspa?section=all for more info
+ * @param {string} html html to be parsed
+ * @returns {string} markdown string
+ */
+export function htmlToJiraMarkdown(html) {
+  let markdown = html;
+
+  // Parse text formats
+  markdown = markdown.replace(/<strong>(.*)<\/strong>/g, (match, p1) => {
+    return `*${p1}*`;
+  });
+  markdown = markdown.replace(/<em>(.*)<\/em>/g, (match, p1) => {
+    return `_${p1}_`;
+  });
+  markdown = markdown.replace(/<u>(.*)<\/u>/g, (match, p1) => {
+    return `+${p1}+`;
+  });
+  markdown = markdown.replace(/<s>(.*)<\/s>/g, (match, p1) => {
+    return `-${p1}-`;
+  });
+
+  // Parse indentations
+  markdown = markdown.replace(/<p class="ql-indent-1">(.*)<\/p>/g, (match, p1) => {
+    return `\t${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<p class="ql-indent-2">(.*)<\/p>/g, (match, p1) => {
+    return `\t\t${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<p class="ql-indent-3">(.*)<\/p>/g, (match, p1) => {
+    return `\t\t\t${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<p class="ql-indent-4">(.*)<\/p>/g, (match, p1) => {
+    return `\t\t\t\t${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<p class="ql-indent-5">(.*)<\/p>/g, (match, p1) => {
+    return `\t\t\t\t\t${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<p class="ql-indent-6">(.*)<\/p>/g, (match, p1) => {
+    return `\t\t\t\t\t\t${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<p class="ql-indent-7">(.*)<\/p>/g, (match, p1) => {
+    return `\t\t\t\t\t\t\t${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<p class="ql-indent-8">(.*)<\/p>/g, (match, p1) => {
+    return `\t\t\t\t\t\t\t\t${p1}\r\n`;
+  });
+
+  // Parse headings
+  markdown = markdown.replace(/<h1>(.*)<\/h1>/g, (match, p1) => {
+    return `h1. ${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<h2>(.*)<\/h2>/g, (match, p1) => {
+    return `h2. ${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<h3>(.*)<\/h3>/g, (match, p1) => {
+    return `h3. ${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<h4>(.*)<\/h4>/g, (match, p1) => {
+    return `h4. ${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<h5>(.*)<\/h5>/g, (match, p1) => {
+    return `h5. ${p1}\r\n`;
+  });
+  markdown = markdown.replace(/<h6>(.*)<\/h6>/g, (match, p1) => {
+    return `h6. ${p1}\r\n`;
+  });
+
+  // Parse links
+  markdown = markdown.replace(
+    /<a href="(.*)" rel="noopener noreferrer" target="_blank">(.*)<\/a>/g,
+    (match, p1, p2) => {
+      return `[${p2}|${p1}]`;
+    },
+  );
+
+  // Parse code blocks
+  markdown = markdown.replace(/<code>(.*)<\/code>/g, (match, p1) => {
+    return `{code}${p1}{code}`;
+  });
+  markdown = markdown.replace(/<pre>(.*)<\/pre>/g, (match, p1) => {
+    return `{code}${p1}{code}`;
+  });
+
+  // TODO: Parse lists
```

Comment:
In TSSW all TODOs require a DM ticket. Please create one and add the ticket ID here.

Reply:
A new learning for the day :muscle:. Thanks @wvreeven ! I've seen TODOs referencing DM tickets before, but didn't know it was a standard. I'll make the change :+1:.